dev/test/commands/generate_docs_command_test.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "tmpdir"
+
+module ProductTaxonomy
+  class GenerateDocsCommandTest < TestCase
+    setup do
+      @tmp_path = Dir.mktmpdir
+      @original_data_path = ProductTaxonomy::DATA_PATH
+      ProductTaxonomy.send(:remove_const, :DATA_PATH)
+      ProductTaxonomy.const_set(:DATA_PATH, @tmp_path)
+
+      # Create test files
+      FileUtils.mkdir_p(File.expand_path("../dist/en/integrations", @tmp_path))
+      File.write(
+        File.expand_path("../dist/en/integrations/all_mappings.json", @tmp_path),
+        JSON.fast_generate({
+          "mappings" => [{
+            "input_taxonomy" => ["shopify"],
+            "output_taxonomy" => ["external"],
+            "rules" => [{
+              "input" => { "category" => "test" },
+              "output" => { "category" => ["mapped"] },
+            }],
+          }],
+        }),
+      )
+
+      # Create template files
+      FileUtils.mkdir_p(File.expand_path("../docs/_releases", @tmp_path))
+      File.write(
+        File.expand_path("../docs/_releases/_index_template.html", @tmp_path),
+        "TITLE TARGET GH_URL",
+      )
+      File.write(
+        File.expand_path("../docs/_releases/_attributes_template.html", @tmp_path),
+        "TITLE TARGET GH_URL",
+      )
+
+      Command.any_instance.stubs(:load_taxonomy)
+
+      Serializers::Category::Docs::SiblingsSerializer.stubs(:serialize_all).returns({ "siblings" => "foo" })
+      Serializers::Category::Docs::SearchSerializer.stubs(:serialize_all).returns([{ "search" => "bar" }])
+      Serializers::Attribute::Docs::BaseAndExtendedSerializer.stubs(:serialize_all).returns({ "attributes" => "baz" })
+      Serializers::Attribute::Docs::ReversedSerializer.stubs(:serialize_all).returns({ "reversed" => "qux" })
+      Serializers::Attribute::Docs::SearchSerializer.stubs(:serialize_all).returns([{ "attribute_search" => "quux" }])
+    end
+
+    teardown do
+      FileUtils.remove_entry(@tmp_path)
+      ProductTaxonomy.send(:remove_const, :DATA_PATH)
+      ProductTaxonomy.const_set(:DATA_PATH, @original_data_path)
+    end
+
+    test "initialize sets version to unstable by default" do
+      command = GenerateDocsCommand.new({})
+      assert_equal "unstable", command.instance_variable_get(:@version)
+    end
+
+    test "initialize accepts custom version" do
```

Comment:
Should we have a test for poorly formatted versions?

Reply:
Done